db/db_impl/db_spdb_impl_write.cc

```
@@ -0,0 +1,548 @@
+    } else {
+      uint64_t progress_batch_seq;
+      size_t wal_writes = 0;
+      WriteBatch* merged_batch = &tmp_batch_;
+      for (const WriteBatch* batch : batch_group->wal_writes_) {
+        if (wal_writes != 0 &&
+            (progress_batch_seq != WriteBatchInternal::Sequence(batch))) {
+          // this can happened if we have a batch group that consists no wal
+          // writes... need to divide the wal writes when the seq is broken
+          io_s =
+              db_->SpdbWriteToWAL(merged_batch, wal_writes, to_be_cached_state);
+          // reset counter and state
+          tmp_batch_.Clear();
+          wal_writes = 0;
+          to_be_cached_state = nullptr;
+          if (!io_s.ok()) {
+            // TBD what todo with error
+            break;
+          }
+        }
+        if (wal_writes == 0) {
+          // first batch seq to use when we will replay the wal after recovery
+          WriteBatchInternal::SetSequence(merged_batch,
+                                          WriteBatchInternal::Sequence(batch));
+        }
+        // to be able knowing the batch are in seq order
+        progress_batch_seq =
+            WriteBatchInternal::Sequence(batch) + batch->Count();
+        Status s = WriteBatchInternal::Append(merged_batch, batch, true);
+        // Always returns Status::OK.()
+        if (!s.ok()) {
+          assert(false);
+        }
+        if (WriteBatchInternal::IsLatestPersistentState(batch)) {
+          // We only need to cache the last of such write batch
+          to_be_cached_state = batch;
+        }
+        ++wal_writes;
+      }
+      if (wal_writes) {
+        io_s =
+            db_->SpdbWriteToWAL(merged_batch, wal_writes, to_be_cached_state);
+        tmp_batch_.Clear();
+      }
+    }
+    if (!io_s.ok()) {
+      // TBD what todo with error
+      ROCKS_LOG_ERROR(db_->immutable_db_options().info_log,
+                      "Error write to wal!!! %s", io_s.ToString().c_str());
```

This is not enough. We're outputting the error to log, but we're swallowing the error here, and writes will still get an OK status from their `Write()` call even though the data hasn't been persisted.